### ISSUE TITLE
Different constructors can be equal

### ISSUE THREAD

**Issue by radix** [podhmo/miniadt#2]
e.g.

``` python
from miniadt import ADTTypeProvider

List = ADTTypeProvider('List')
Cons = List('Cons', 'head tail')
Cons2 = List('Cons2', 'head tail')
Nil = List('Nil', '')

Cons(1, Nil()) == Cons2(1, Nil()) # returns True, which is unexpected
```

likewise for completely different types:

``` python
T1 = ADTTypeProvider('T1')
C1 = T1('C1', '')
T2 = ADTTypeProvider('T2')
C2 = T2('C2', '')
C1() == C2() # returns True, which is unexpected
```

In all ADT systems that I know, different constructors are not comparable at all, so I would suggest that these comparisons should return False.

**Comment by podhmo** [comment]
oh, i agree.

**Comment by podhmo** [comment]
namedtuple is just tuple.
I didn't recognize it when making this package.

```
>>> from collections import namedtuple
>>> A = namedtuple("A", "x")
>>> B = namedtuple("B", "y")
>>> A(10) == B(10)
True
```

**Comment by podhmo** [comment]
I have fixed this issue with #1 's  approach. 
adding `adttype` function, this is generating class definition code, on the fly, and exec it.
https://github.com/podhmo/miniadt/commit/6f7dc19e177a1e53c67a1885754ebee32cdcf125
